src/rdkafka_conf.c

```diff
@@ -678,6 +678,12 @@ static const struct rd_kafka_property rd_kafka_properties[] = {
           _RK(ssl.keystore_location),
           "Path to client's keystore (PKCS#12) used for authentication."
         },
+#if OPENSSL_VERSION_NUMBER >= 0x10100000
+        { _RK_GLOBAL, "openssl.engine.location", _RK_C_STR,
+          _RK(ssl.openssl_engine_location),
+          "Path to openssl engine dll. OpenSSL >= 1.1.0 required."
```

Comment:
It should probably say that it loads the engine's certificates by default, or whatever it does.
What happens if keys, certs and CAs are manually configured too? Will they be overwritten with whatever the engine provides?
What about the CA certs loaded from the Root store?

Reply:
sure will change the message to call it library.

Since most of the security is being built around ssl certificates and CA are still accessible, I am not extending engine support to load CA certs. CA should continue to be manually configured using existing ways OR automatically loaded from root store.

Since this is a new feature and application is setting appropriate engine params then YES ssl certs will be replace any private/public keys already set. This is as per the documentation [link](https://www.openssl.org/docs/man1.0.2/man3/SSL_CTX_use_PrivateKey.html) which says  
> "All of the functions to set a new certificate will replace any existing certificate of the same type that has already been set. Similarly all of the functions to set a new private key will replace any private key that has already been set"